model/custom_status.go

```diff
@@ -33,6 +56,16 @@ func (cs *CustomStatus) ToJson() string {
 	return string(b)
 }
 
+func (cs *CustomStatus) IsDurationValid() bool {
+	return cs.Duration == "" || validCustomStatusDuration[cs.Duration]
+}
+
+func (cs *CustomStatus) IsExpirationTimeValid() bool {
+	isStatusDurationEmpty := cs.Duration == "" || cs.Duration == "dont_clear"
+	isStatusAlreadyExpired := cs.ExpiresAt.IsZero() || cs.ExpiresAt.Before(time.Now())
+	return isStatusDurationEmpty || !isStatusAlreadyExpired
```

Comment:
If I'm reading this correctly, we don't check if `ExpiresAt` is valid when `Duration` is empty. Does that mean I can't just send up `{ExpiresAt: "2021-05-31T00:00:00"}` but /also/ have to specify a `Duration`?

Reply:
Yes, because not sending a duration means never clearing the status.

Reply:
But I'm thinking if `duration` is empty and `expires_at` is not, then we should set `duration` to custom `date_and_time` and set the expiry time. I think this would be better. What do you think?

Reply:
I wonder if we could make a case for never remembering `Duration`, and just always setting `ExpiresAt`? @esethna, is it important that the client remember the last duration I picked for the next emoji I want to set? I feel like it would be more useful to keep this simple and exclusively model `ExpiresAt` server-side.

Reply:
>  @esethna, is it important that the client remember the last duration I picked for the next emoji I want to set

No that's not important in my mind. I'd imagine we reset the expires at time in the UI to the default (tomorrow) for any new custom statuses set

Reply:
The field `Duration` is needed for the recent custom statuses that we show in the modal along with the suggestions. Without this field, we will not be able to use the recent statuses with the clear time as defined in the [spec](https://mattermost.atlassian.net/wiki/spaces/CS/pages/983302547/User+Experience+and+Functional+Notes).

Reply:
Thanks for the extra context, @manojmalik20! Indeed, `Duration` proves useful as you describe, and a `date_and_time` option for when `ExpiresAt` is used suffices.